File: JS/twosum.js
let nums = [2,7,11,15], target = 9;

let map = new Map();
for(let i=0; i<nums.length; i++){
    let otherNum = target - nums[i];
    if(map.has(otherNum)){
        console.log([map.get(otherNum), i]);
    }
    map.set(nums[i],i);
}
console.log("end");

File: JavaWork/twosum.java
import java.util.*;

public class twosum {
    public static void main(String[] args) {
        int nums[] = {3,3}, target = 6;
        Map<Integer, Integer> map = new HashMap<>();

        for(int i=0; i<nums.length; i++){
            int otherNum = target - nums[i];
            if(map.containsKey(otherNum)){
                int res[] = {map.get(otherNum), i};
                System.out.println(Arrays.toString(res));
                // return result;
            } else {
                map.put(nums[i], i);
            }
        }

    }
}
